smack-extensions/src/main/java/org/jivesoftware/smackx/dataformmedia/MediaElement.java

```diff
@@ -0,0 +1,100 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.dataformmedia;
+
+import java.util.List;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.packet.XmlEnvironment;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+
+public final class MediaElement implements ExtensionElement {
+
+    private String NAMESPACE = "urn:xmpp:media-element";
+    private String ELEMENT = "media";
+    private final List<URINode> uriNodeList;
+    private final String attribute_height = "height";
+    private final String attribute_width = "width";
+    private final Integer height;
+    private final Integer width;
+
+    private MediaElement(List<URINode> uriNodeList) {
+        this.uriNodeList = uriNodeList;
+        this.height = null;
+        this.width = null;
+    }
+
+    private MediaElement(List<URINode> uriNodeList, int height, int width) {
+        this.uriNodeList = uriNodeList;
+        this.height = height;
+        this.width = width;
+    }
+
+    @Override
+    public String getNamespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public String getElementName() {
+        return ELEMENT;
+    }
+
+    @Override
+    public CharSequence toXML(XmlEnvironment xmlEnvironment) {
+        XmlStringBuilder xml = new XmlStringBuilder(this);
+        if (height != null && width != null) {
+            xml.attribute(attribute_height, Integer.toString(height));
+            xml.attribute(attribute_width, Integer.toString(width));
+        }
+        xml.rightAngleBracket();
+        xml.append(getMediaXML());
+        return xml.closeElement(getElementName());
+    }
+
+    private CharSequence getMediaXML() {
+        XmlStringBuilder mediaXML = new XmlStringBuilder();
+        for (URINode node : uriNodeList) {
+            mediaXML.element(node);
+        }
+        return mediaXML;
+    }
+
+    public static Builder getBuilder() {
+        return new Builder();
+    }
+
+    public static final class Builder{
```

Comment:
Not sure if the Builder pattern is useful here, as `build(...)` is the only method. Making the constructors public would probably be sufficient in this situation.